traffic_ops/traffic_ops_golang/topology/topologies.go

```diff
@@ -563,6 +614,23 @@ JOIN topology_cachegroup tc on t.name = tc.topology
 	return query
 }
 
+func selectEmptyCacheGroupsQuery() string {
+	// language=SQL
+	query := `
+		SELECT c."name", COUNT((
+			SELECT s2.id FROM server s2
+			JOIN "type" t ON s."type" = t.id
+			WHERE s2.id = s.id
+		)) server_count
+		FROM cachegroup c
+		LEFT JOIN "server" s ON c.id = s.cachegroup
+		WHERE c."name" = ANY(CAST(:cachegroup_names AS TEXT[]))
+		GROUP BY c."name"
+		ORDER BY server_count
```

Comment:
Can this simply be
```
SELECT c."name", COUNT(*) AS server_count
        FROM cachegroup c
        LEFT JOIN "server" s ON c.id = s.cachegroup
        WHERE c."name" = ANY(CAST(:cachegroup_names AS TEXT[]))
        GROUP BY c."name"
        ORDER BY server_count;
```
instead?

Reply:
Due to the `LEFT JOIN` making `COUNT(*)` include `NULL`s, it would need to be something like `COUNT(*) FILTER (WHERE s.id IS NOT NULL)`.

Simplified in 118863a5e